### PR TITLE
remove legacy dc cache methods

### PR DESCRIPTION
**Pull request by deads2k**
Switches the DC controllers to shared informers and listers instead of their one offs.

@mfojtik this is part of what is under `pkg/client`

### CODE REVIEW

File: pkg/apps/generated/listers/apps/internalversion/deploymentconfig_expansion.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		return nil, errors.NewNotFound(apps.Resource("deploymentconfig"), dcName)
+	}
+	return obj.(*apps.DeploymentConfig), nil
+}
+
+// GetConfigsForImageStream returns all the deployment configs that point to the provided image stream
+// by searching through using the ImageStreamReferenceIndex (deployment configs are indexed in the cache
+// by namespace and by image stream references).
+func (s *deploymentConfigLister) GetConfigsForImageStream(stream *imageapi.ImageStream) ([]*apps.DeploymentConfig, error) {
+	items, err := s.indexer.ByIndex(ImageStreamReferenceIndex, stream.Namespace+"/"+stream.Name)
```
Comment (deads2k) [inline pkg/apps/generated/listers/apps/internalversion/deploymentconfig_expansion.go]: This index was never added before, so all calls to this method in the legacy package returned an error!

File: pkg/apps/controller/deploymentconfig/factory.go [hunk 2]
```diff
@@ -202,12 +204,17 @@ func (c *DeploymentConfigController) work() bool {
 	}
 	defer c.queue.Done(key)
 
-	dc, err := c.getByKey(key.(string))
+	namespace, name, err := cache.SplitMetaNamespaceKey(key.(string))
 	if err != nil {
 		utilruntime.HandleError(err)
+		return false
```
Comment (mfojtik) [inline pkg/apps/controller/deploymentconfig/factory.go]: we don't care about wrong keys?

Comment (mfojtik) [reply pkg/apps/controller/deploymentconfig/factory.go]: nvmd, i thought you removed the handle error...

File: pkg/cmd/server/start/informers.go [hunk 3]
```diff
@@ -29,6 +30,7 @@ import (
 	templateclient "github.com/openshift/origin/pkg/template/generated/internalclientset"
 	userinformer "github.com/openshift/origin/pkg/user/generated/informers/internalversion"
 	userclient "github.com/openshift/origin/pkg/user/generated/internalclientset"
+	"k8s.io/client-go/tools/cache"
```
Comment (mfojtik) [inline pkg/cmd/server/start/informers.go]: move this import up

**Comment by mfojtik** [comment]
one minor nit about import but this LGTM

**Comment by deads2k** [comment]
comments addressed.

**Pull request merged**